File: test/export/pas/07-subprocesos.pas
Program sin_titulo;
Type
	SIN_TIPO = Real;
	TIPO_VECTOR_10 = Array [1..10] Of SIN_TIPO;
	TIPO_MATRIZ_3X4 = Array [1..3, 1..4] Of SIN_TIPO;
Var
	a: Array [1..10] Of SIN_TIPO;
	b: Array [1..3,1..4] Of SIN_TIPO;
	c: Real;
Function funciondoble(b: Real): Real;
Begin
	funciondoble := 2*b;
End;
Procedure noretornanada(a: SIN_TIPO; b: SIN_TIPO);
Begin
	WriteLn(a+b);
End;
Procedure porreferencia(Var b: Real);
Begin
	b := 7;
End;
Procedure recibevector(Var v: TIPO_VECTOR_10);
Begin
	ReadLn(v[1]);
End;
Procedure recibematriz(Var m: TIPO_MATRIZ_3X4);
Begin
	ReadLn(m[1,1]);
End;
Begin
	WriteLn(funciondoble(5));
	noretornanada(3,9);
	c := 0;
	porreferencia(c);
	WriteLn(c);
	recibevector(a);
	recibematriz(b);
End.

File: test/export/m/07-subprocesos.m
function sin_titulo()
	disp(funciondoble(5));
	noretornanada(3,9);
	c=0;
	porreferencia(c);
	disp(c);
	a=zeros(1,10);
	b=zeros(3,4);
	recibevector(a);
	recibematriz(b);
end
function a = funciondoble(b)
	a=2*b;
end
function noretornanada(a, b)
	disp(a+b);
end
function porreferencia(b)
	b=7;
end
function recibevector(v)
	v(1)=input('');
end
function recibematriz(m)
	m(1,1)=input('');
end
